esp-hal-common/src/spi_slave.rs

```diff
@@ -0,0 +1,1099 @@
+            read_buffer_ptr,
+            read_buffer_len,
+        )?;
+
+        self.clear_dma_interrupts();
+        reset_dma_before_usr_cmd(reg_block);
+
+        // On the esp32, all full-duplex transfers are single, and all half-duplex
+        // transfers use the cmd/addr/dummy/data sequence (but are still
+        // single).
+        #[cfg(not(esp32))]
+        reg_block
+            .dma_conf
+            .modify(|_, w| w.dma_slv_seg_trans_en().clear_bit());
+
+        tx.start_transfer()?;
+        Ok(rx.start_transfer()?)
+    }
+
+    fn start_write_bytes_dma(
+        &mut self,
+        ptr: *const u8,
+        len: usize,
+        tx: &mut TX,
+    ) -> Result<(), Error> {
+        let reg_block = self.register_block();
+
+        tx.is_done();
+
+        self.enable_dma();
+
+        reset_dma_before_load_dma_dscr(reg_block);
+        tx.prepare_transfer_without_start(self.dma_peripheral(), false, ptr, len)?;
+
+        self.clear_dma_interrupts();
+        reset_dma_before_usr_cmd(reg_block);
+
+        // On the esp32, all full-duplex transfers are single, and all half-duplex
+        // transfers use the cmd/addr/dummy/data sequence (but are still
+        // single).
+        #[cfg(not(esp32))]
+        reg_block
+            .dma_conf
+            .modify(|_, w| w.dma_slv_seg_trans_en().clear_bit());
+
+        Ok(tx.start_transfer()?)
+    }
+
+    fn start_read_bytes_dma(&mut self, ptr: *mut u8, len: usize, rx: &mut RX) -> Result<(), Error> {
+        let reg_block = self.register_block();
+
+        rx.is_done();
+
+        self.enable_dma();
+
+        reset_dma_before_load_dma_dscr(reg_block);
+        rx.prepare_transfer_without_start(false, self.dma_peripheral(), ptr, len)?;
+
+        self.clear_dma_interrupts();
+        reset_dma_before_usr_cmd(reg_block);
+
+        // On the esp32, all full-duplex transfers are single, and all half-duplex
+        // transfers use the cmd/addr/dummy/data sequence (but are still
+        // single).
+        #[cfg(not(esp32))]
+        reg_block
+            .dma_conf
+            .modify(|_, w| w.dma_slv_seg_trans_en().clear_bit());
+
+        Ok(rx.start_transfer()?)
+    }
+
+    fn dma_peripheral(&self) -> DmaPeripheral {
+        match self.spi_num() {
+            2 => DmaPeripheral::Spi2,
+            #[cfg(any(esp32, esp32s2, esp32s3))]
+            3 => DmaPeripheral::Spi3,
+            _ => panic!("Illegal SPI instance"),
+        }
+    }
+
+    #[cfg(any(esp32c2, esp32c3, esp32c6, esp32h2, esp32s3))]
+    fn enable_dma(&self) {
+        let reg_block = self.register_block();
+        reg_block.dma_conf.modify(|_, w| {
+            w.dma_tx_ena()
+                .set_bit()
+                .dma_rx_ena()
+                .set_bit()
+                .rx_eof_en()
+                .clear_bit()
+        });
+    }
+
+    #[cfg(any(esp32, esp32s2))]
+    fn enable_dma(&self) {
+        // for non GDMA this is done in `assign_tx_device` / `assign_rx_device`
```

Comment:
I'm just skimming through stuff, but this is very suspicious to me. Github can't find either `assign_tx_device` or `assign_rx_device`. Maybe DMA is just simply not enabled on ESP32 and S2?